File: src/strategy/examples/searchintent/VerifyLossSearchStrategy.java
package strategy.examples.searchintent;

import java.util.List;

import abstractfactory.examples.searchcatalog.CatalogFactory;
import adapter.examples.metadata.ImageMetadata;

/** RF2.5: verify loss -> match against active lost-pet alerts. */
public class VerifyLossSearchStrategy implements SearchStrategy {

    private final CatalogFactory factory;

    public VerifyLossSearchStrategy(CatalogFactory factory) { this.factory = factory; }

    @Override
    public SearchResult search(ImageMetadata meta) {
        List<String> matches = factory.createLostAlertsCatalog().query(meta);
        return new SearchResult(SearchIntent.VERIFY_LOSS, matches);
    }
}

File: src/strategy/examples/searchintent/VerifyLossSearchStrategy.js
'use strict';

const SearchIntent = require('./SearchIntent');
const SearchResult = require('./SearchResult');

/** RF2.5: verificar perdida -> match contra alertas activas. */
class VerifyLossSearchStrategy {
  constructor(factory) { this.factory = factory; }
  search(meta) {
    const matches = this.factory.createLostAlertsCatalog().query(meta);
    return new SearchResult(SearchIntent.VERIFY_LOSS, matches);
  }
}

module.exports = VerifyLossSearchStrategy;
